### PR TITLE
Traits for ABS

### PR DESCRIPTION
**Pull request by Edkamb**

### CODE REVIEW

**Comment by VolkerStolz** [comment]
@rm2340 will quickly go over it with Eduard and they'll let us know when we should hit the MERGE button.

**Comment by rudi** [comment]
I'm in communication with Eduard via email as well, want to check out a few things next week before merging. Great if Radu can have a look as well!

File: abs-docs/asciidoc/oo.adoc [hunk 1]
```diff
@@ -74,9 +74,10 @@ block has the same restrictions as in the init block.
 .Syntax
 |====
 | _ClassDecl_ ::= | `class` _SimpleTypeIdentifier_ [ `(` [ _ClassParameterList_ ] `)` ] [ `implements` _InterfaceList_ ]
-|                 | `{` [ _FieldDeclList_ ] [ _Block_ ] [_RecoveryBlock_] [ _MethDeclList_ ] `}`
+|                 | `{` [ _TraitUseList_ ] [ _FieldDeclList_ ] [ _Block_ ] [_RecoveryBlock_] [ _MethDeclList_ ] `}`
 | _ClassParameterList_ ::= | _Type_ _SimpleIdentifier_ { `,` _Type_ _SimpleIdentifier_ }
 | _InterfaceList_ ::= | _TypeIdentifier_ { `,` _TypeIdentifier_ }
+| _TraitUseList_ ::= | adds _TraitName_`;` { `,` adds _TraitName_ `;`}
```
Comment (rudi) [inline abs-docs/asciidoc/oo.adoc]: Need a space between _ and ` in _TraitName`;`
Also, the comma `,` is not in the grammar as defined in ABS.g4

**Comment by rudi** [comment]
Double-merged via command line and website :)

**Pull request merged**